packages/cli/src/oclif/commands/test.js

```diff
@@ -31,20 +34,33 @@ class TestCommand extends BaseCommand {
     }
 
     const env = Object.assign({}, process.env, extraEnv);
+
+    const useYarn =
+      this.flags.yarn ||
+      (await pathExists(path.join(process.cwd(), 'yarn.lock')));
```

Comment:
not really, no! just tending more towards async code over sync when possible ¯\\\_(ツ)_/¯

The implementation is pretty simple: 

https://github.com/jprichardson/node-fs-extra/blob/7498c9c9ebe5e83e535092aedb9aad3e23f601bf/lib/path-exists/index.js#L6

Reply:
Since here we're doing `await` anyway, I thought using an async function didn't bring benefits. But not a big deal, I'm fine with either way.

Reply:
> Since here we're doing await anyway, I thought using an async function didn't bring benefits

This is an unexpectedly philosophical question that I'm not finding a great answer for 😅 

In a situation where the process is doing a lot of things at once (like a webserver), the `await` lets the process work elsewhere (like on a different request) while it's waiting. For our purposes, the CLI probably isn't doing a lot of other work at the time, so i'm not sure it does much for us here specifically.

If nothing else, I think it makes sense to not block if you don't have to. ¯\\\_(ツ)_/¯ I wish I knew more here though!